homeassistant/components/zha/light.py

```diff
@@ -203,6 +206,10 @@ def async_restore_last_state(self, last_state):
         brightness = kwargs.get(light.ATTR_BRIGHTNESS)
         effect = kwargs.get(light.ATTR_EFFECT)
 
+        if brightness is None and self._off_brightness is not None:
+            brightness = self._off_brightness
+        self._off_brightness = None
```

Comment:
Should we reset it only if the service call was successful? ie move it to https://github.com/home-assistant/home-assistant/blob/ab979e72d727941b96b86ee5dcdf1d7ab5a4abd0/homeassistant/components/zha/light.py#L295